haystack_experimental/components/tools/openapi/_schema_conversion.py

```diff
@@ -0,0 +1,277 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+import jsonref
+
+MIN_REQUIRED_OPENAPI_SPEC_VERSION = 3
+
+logger = logging.getLogger(__name__)
+
+
+def openai_converter(schema: "OpenAPISpecification") -> List[Dict[str, Any]]:  # type: ignore[name-defined] # noqa: F821
```

Comment:
I think you should be able to remove the `type: ignore` by using the following code:
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
	# import OpenAPISpecification; won't cause circular imports
```

Reply:
@shadeMe it didn't help - I can move `Operation` and `OpenAPISpecification` to `types.py` - that works nicely and further simplifies `_openapi.py`